### ISSUE TITLE
Fallback for a Fallback

### ISSUE THREAD

**Issue by sumitsharma0819** [Netflix/Hystrix#1744]
I am trying to write a logic where
a) I will call Service2 if Service1 fails.
b) And in case Service2 fails too, call Service3.

For Step a, I can use the fallback. So the question is can a fallback (Service2) have its own fallback (Service3)?

Is this correct approach?

**Comment by sriece91** [comment]
No. Whenever you are trying to emit fallback of a service (say Service1) as an another command object (say Service2), the respective run() method in the Service2 class should be having no possibilities to throw further exceptions or errors. It should only return respective message and avoid doing any other operations in most of the cases. Even if you try do something excess, handle with care.

Chaining of so many Services particularly for fallback is not a good way and it leads to creation and deletion of so many threads thus wasting system resources. Hence, keeping the run() method of Service2 class to return simply the exception/error message is a good way.

**Comment by mattnelson** [comment]
You can nest circuits.

https://github.com/Netflix/Hystrix/wiki/How-To-Use#Common-Patterns-FallbackCacheViaNetwork
